include/xtensor/xoperation.hpp

```diff
@@ -109,14 +116,14 @@ namespace xt
 
             template <class A2, class A3>
             constexpr auto operator()(bool t1, const A2& t2, const A3& t3) const noexcept
-                -> disable_at_least_one_xoptional<A2, A3, std::common_type_t<A2, A3>>
+                -> std::enable_if_t<!at_least_one_xoptional<A2, A3>::value, std::common_type_t<A2, A3>>
```

Comment:
Would be more readable to use the same pattern as the other overload, with XTL_REQUIRES

Reply:
I totally agree, thanks